kth-largest-element-in-a-stream/step1.py

```diff
@@ -0,0 +1,44 @@
+from heapq import heappush, heappop
+from typing import List
+
+# [], -3, -2, -4, 0, 4
+# 1
+
+#           1, 2, 3,   4,      5,   6,      7,   8,     9,  10,   11
+# stream    []    [-3] [-3,-2] [-2] [-4,-2] [-2] [-2,0] [0] [0,4] [4]
+# removable    -1
+
+class KthLargest:
+
+    def __init__(self, k: int, nums: List[int]):
+        self.k = k
+        self.stream = []
+        for value in nums:
+            heappush(self.stream, value)
+        removable = len(nums) - k
+        if removable <= 0:
+            return
+        for _ in range(removable):
+            heappop(self.stream)
```

Comment:
このあたりは「numsを降順ソートし先頭k個を取り、それをheapifyする」というやり方でシンプルに書く方法もあります（別途heapifyをimportする必要があります）。
```suggestion
        self.stream = sorted(nums, reverse=True)[:k]
        heapify(self.stream)
```